bundles/framework/guidedtour/handler/GuidedTourHandler.js

```diff
@@ -0,0 +1,192 @@
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+import { showGuidedTourPopup } from '../view/GuidedTourPopup';
+
+class TourHandler extends StateHandler {
+    constructor (instance) {
+        super();
+        this.instance = instance;
+        this.sandbox = Oskari.getSandbox();
+        this.setState({
+            step: 0,
+            steps: [],
+            position: null,
+            loading: false
+        });
+        this.popupControls = null;
+        this._localization = Oskari.getLocalization('GuidedTour');
+        this.initSteps();
+    };
+
+    getName () {
+        return 'GuidedTourHandler';
+    }
+
+    initSteps () {
+        const delegate = {
+            bundleName: this.instance.getName(),
+            priority: 0,
+            getTitle: () => {
+                return this._localization.page1.title;
+            },
+            getContent: () => {
+                return this._localization.page1.message;
+            }
+        };
+        this.addStep(delegate);
+    }
+
+    addStep (delegate) {
+        if (this.instance.conf && this.instance.conf.steps) {
+            // step ordering
+            const stepSpec = this.instance.conf.steps;
+            const index = stepSpec.map((s) => s.bundleName).indexOf(delegate.bundleName);
+            if (delegate.bundleName !== this.instance.getName()) {
+                if (index < 0) {
+                    return;
+                }
+                delegate.priority = index + 1;
+            }
+
+            // custom content
+            if (index >= 0) {
+                const content = stepSpec[index].content;
+                if (content) {
+                    delegate.getContent = () => { // empty placeholder while loading
+                        return '';
+                    };
+                    this.getGuideContent(content, (success, response) => {
+                        if (success) {
+                            delegate.getContent = () => { return response.body; };
+                            delegate.getTitle = () => { return response.title; };
+                        } else {
+                            Oskari.log(this.instance.getName()).error('Failed to load guided tour content for step "' + stepSpec[index].bundleName + '" with tags: ' + content);
+                        }
+                    });
+                }
+            }
+        }
+        if (typeof delegate.priority === 'number') {
+            const priorities = this.state.steps.map((d) => d.priority);
+            const insertLocation = _.sortedIndex(priorities, delegate.priority);
```

Comment:
Replace lodash with something else since we are trying to get rid of it at some point.